src/recipes/modal.ts

```diff
@@ -0,0 +1,162 @@
+      border: 'none',
+      display: 'block',
+      width: '100%',
+      padding: '8%',
+      boxSizing: 'border-box',
+      zIndex: 'popup',
+      backgroundColor: 'bg',
+      transition: 'all 0.4s ease-out',
+    },
+    title: {
+      //   fontWeight: 400, from
+      fontWeight: 700,
+      marginBottom: '40px',
+      textAlign: 'center',
+    },
+    text: {
+      marginBottom: '2em',
+    },
+    actions: {
+      textAlign: 'center',
+    },
+    content: {
+      maxWidth: '40em',
+      margin: '0 auto',
+      maxHeight: 'none',
+    },
+    close: {
+      position: 'absolute',
+      top: '-5px',
+      right: '-5px',
+      fontSize: '12px',
+      verticalAlign: 'middle',
+      textAlign: 'center',
+      padding: '10px',
+      display: 'none',
+      '& span': {
+        display: 'inline-block',
+        width: '11px',
+        height: '11px',
+        color: {
+          base: 'rgba(0, 0, 0, 0.3)',
+          _dark: 'rgba(255, 255, 255, 0.3)',
+        },
+        borderColor: {
+          base: 'rgba(0, 0, 0, 0.3)',
+          _dark: 'rgba(255, 255, 255, 0.3)',
+        },
+      },
+    },
+    closeX: {
+      /* extend click area */
+      padding: '10px 20px',
+      margin: '-10px -20px',
+      position: 'fixed',
+      top: 'calc(9px - 0.2em)',
+      right: '11px',
+      color: 'inherit',
+      textDecoration: 'none',
+    },
+  },
+  variants: {
+    id: {
+      welcome: {
+        root: { height: '100%', fontSize: '26px', paddingTop: '2em' },
+        title: { marginBottom: '0', lineHeight: 1 },
+        text: {
+          marginTop: '2em',
+          marginBottom: '2.5em',
+          fontSize: '75%',
+        },
+        actions: { fontSize: '0.75em' },
+      },
+      home: {
+        // unused, from .modal-home, not sure what this is for
+        root: {
+          marginLeft: '20px',
+          marginTop: '-5px',
+        },
+      },
+      auth: {},
+      customizeToolbar: {},
+      devices: {},
+      export: {},
+      feedback: {},
+      invites: {},
+      help: {},
+      settings: {},
+      signup: {},
+      testGestureDiagram: {},
+    },
+    stack: {
+      true: {
+        actions: {
+          textAlign: 'center',
+          '@media screen and (min-width: 480px)': {
+            display: 'flex',
+            justifyContent: 'center',
+            alignItems: 'center',
+            '& > *': {
+              marginTop: '1em',
+            },
+            '& > :not(:last-child)': {
+              marginRight: '1em',
+            },
+          },
+        },
+      },
+    },
+    center: {
+      true: {
+        root: {
+          textAlign: 'center',
+          left: '0',
+          right: '0',
+          margin: '0 auto',
+        },
+      },
+    },
+    opaque: {
+      true: {
+        root: {
+          animation: 'fadein 0.4s',
+        },
+      },
+    },
+  },
+  jsx: ['Devices', 'Help', 'ShortcutTable', 'Auth', 'ModalComponent'],
+  staticCss: ['*'],
```

Comment:
Can you explain what `staticCss` does? I've read the [panda docs](https://panda-css.com/docs/guides/dynamic-styling#using-static-css) but I don't totally understand.

Reply:
For recipes, PandaCSS only generates the variants that are being used. And sometimes it can't properly detect whether or not it's being used. For example, since the `center` variant in modal is controlled by a variable, PandaCSS couldn't detect that it was being used, so I had to add `staticCss: ['*']` to force it to generate the `center` class. I could have been more specific in `staticCss` and only named the specific problematic variants, but I figured that almost all variants were being used anyway, and there were multiple variants that used variables, so I might as well generate them all.